src/components/ControlPanel.tsx

```diff
@@ -173,47 +158,25 @@ Mru.propTypes = {
 };
 
 const ControlPanel = () => {
-    const fileRegionSize = useSelector(getFileRegions)?.length;
     const mruFiles = useSelector(getMruFiles);
-    const autoRead = useSelector(getAutoRead);
-    const autoReset = useSelector(getAutoReset);
-    const targetIsWritable = useSelector(getIsWritable);
-    const targetIsRecoverable = useSelector(getIsRecoverable);
+
     const targetIsMemLoaded = useSelector(getIsMemLoaded);
     const isProtected = !!useSelector(getDeviceInfo)?.cores.find(
         c => c.protectionStatus !== 'NRFDL_PROTECTION_STATUS_NONE'
     );
     const targetIsReady = useSelector(getIsReady);
     const isJLink = useSelector(getTargetType) === CommunicationType.JLINK;
-    const isUsbSerial =
-        useSelector(getTargetType) === CommunicationType.USBSDFU;
     const isMcuboot = useSelector(getIsMcuboot);
-    const isModem = useSelector(getIsModem);
-    const zipFile = useSelector(getZipFilePath);
 
     const dispatch = useDispatch();
     const closeFiles = () => dispatch(fileActions.closeFiles());
     const refreshAllFiles = () => dispatch(fileActions.refreshAllFiles());
-    const toggleAutoRead = () => dispatch(settingsActions.toggleAutoRead());
-    const toggleAutoReset = () => dispatch(settingsActions.toggleAutoReset());
-    const toggleMcuboot = () => dispatch(mcubootTargetActions.toggleMcuboot());
-    const performRecover = () => dispatch(jlinkTargetActions.recover());
-    const performRecoverAndWrite = () =>
-        dispatch(jlinkTargetActions.recoverAndWrite());
+
     const performSaveAsFile = () => dispatch(jlinkTargetActions.saveAsFile());
     const performJLinkRead = () => dispatch(jlinkTargetActions.read());
-    const performReset = () => {
-        if (isUsbSerial) {
-            dispatch(usbsdfuTargetActions.resetDevice());
-        } else {
-            dispatch(jlinkTargetActions.resetDevice());
-        }
-    };
-    const performWrite = () => {
-        // Refresh all files in case that some files have been updated right before write action.
-        refreshAllFiles();
-        dispatch(targetActions.write());
-    };
```

Comment:
Most if not all the removes of this file need to be reverted.

Reply:
This was only to make the programmer simpler after we copied from programmer. Some elf and resolution functionality is still needed, so I can't revert that.